src/release-pr.ts

```diff
@@ -340,4 +406,111 @@ export class ReleasePR {
   addPath(file: string) {
     return ReleasePR.addPathStatic(file, this.path);
   }
+
+  // BEGIN release functionality
+
+  // Override this method to detect the release version from code (if it cannot be
+  // inferred from the release PR head branch)
+  protected async detectReleaseVersionFromCode(): Promise<string | undefined> {
+    return undefined;
+  }
+
+  private async detectReleaseVersion(
+    mergedPR: MergedGitHubPR,
+    branchName: BranchName | undefined
+  ): Promise<string | undefined> {
+    // try from branch name
```

Comment:
I like where you've landed with this functionality. Perhaps we could add one test that explicitly calls out order of operation, i.e., "it('uses version number in branch name over version number in title')".